g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/source/service/impl/SyncGrmSourceServiceImpl.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2019-2022 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.l10n.source.service.impl;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import com.vmware.l10n.source.service.RemoteSyncService;
+import com.vmware.l10n.source.service.SyncGrmSourceService;
+import com.vmware.l10n.utils.DiskQueueUtils;
+import com.vmware.vip.common.l10n.exception.L10nAPIException;
+import com.vmware.vip.common.l10n.source.dto.ComponentSourceDTO;
+
+/**
+ * 
+ * This class used to send the collection source to GRM server
+ *
+ */
+@Service
+public class SyncGrmSourceServiceImpl implements SyncGrmSourceService {
+	
+	private final static Logger LOGGER = LoggerFactory.getLogger(SyncGrmSourceServiceImpl.class);
+	private final static String LOCAL_STR = "local";
+	private final static String GRM_STR = "grm";
+	private final static long REQ_SPLIT_LIMIT = 1024*1024*10;
+
+	/** the path of local resource file,can be configured in spring config file **/
+	@Value("${source.bundle.file.basepath}")
+	private String basePath;
+
+	/** the url of GRM API,can be configured in spring config file **/
+	@Value("${grm.server.url}")
+	private String remoteGRMURL;
+
+	/** the url of Singleton API,can be configured in spring config file **/
+	@Value("${vip.server.url}")
+	private String remoteVIPURL;
+
+	@Autowired
+	private RemoteSyncService remoteSyncService;
+
+	/**
+	 * it represents the remote service is available.
+	 */
+	private boolean grmConnected = false;
+
+	/**
+	 * 
+	 * Synchronize the updated source to remote GRM server
+	 *
+	 */
+	@Override
+	public synchronized void sendSourceToGRM() {
+
+		if (LOCAL_STR.equalsIgnoreCase(remoteGRMURL)) {
+			return;
+		}
+		try {
+			remoteSyncService.ping(remoteGRMURL);
+			setGrmConnected(true);
+			processGRMQueueFiles();
+		} catch (L10nAPIException e) {
+			setGrmConnected(false);
+			LOGGER.error("Remote [" + remoteGRMURL + "] is not connected.", e);
+
+		}
+	}
+	
+    /**
+     * scan the GRM cached source file DIR and process the cached source file
+     */
+	private synchronized void processGRMQueueFiles() {
+        
+		List<File> queueFiles = DiskQueueUtils.listQueueFiles(new File(basePath + DiskQueueUtils.L10N_TMP_GRM_PATH));
+		if (queueFiles == null) {
+			return;
+		}
+		LOGGER.debug("the GRM cache file size---{}", queueFiles.size());
+        
+		for (File quefile : queueFiles) {
+			if(quefile == null || quefile.length()>REQ_SPLIT_LIMIT) {
```

Comment:
what's the condition will trigger the quefile is null?

Reply:
remove it